Functions/Assertions/Should.ps1

```diff
@@ -59,14 +59,15 @@ function Get-FailureMessage($shouldArgs, $value) {
 
     return (& $failureMessageFunction $value $shouldArgs.ExpectedValue)
 }
-function New-ShouldException ($Message,$Line) {
+function New-ShouldException ($Message, $Line, $LineText) {
     $exception = New-Object Exception $Message
     $errorID = 'PesterAssertionFailed'
     $errorCategory = [Management.Automation.ErrorCategory]::InvalidResult
-    $errorRecord = New-Object Management.Automation.ErrorRecord $exception, $errorID, $errorCategory, $null
-    $errorRecord.ErrorDetails = "$Message failed at line: $line"
-
-    $errorRecord
+    # we use ErrorRecord.TargetObject to pass structured information about the error to a reporting system.
+    $targetObject = @{message = $Message; line = $line; linetext = $LineText}
+    $errorRecord = New-Object Management.Automation.ErrorRecord $exception, $errorID, $errorCategory, $targetObject
+    $errorRecord.ErrorDetails = "$Message failed at line: $line : $LineText"
```

Comment:
This line can go away; nothing else is looking at the ErrorDetails property of ShouldExceptions anymore.